swat/cas/connection.py

```diff
@@ -419,66 +419,45 @@ def _get_server_features(self):
         return stype, version, out
 
     def _detect_protocol(self, hostname, port, protocol=None):
-        '''
+        """
         Detect the protocol type for the given host and port
+        :param hostname: string. The CAS host to connect to.
+        :param port: int. The CAS port to connect to.
+        :param protocol: string, optional. The protocol override value.
+        :return: string. 'cas', 'http', or 'https'
+        """
 
-        Parameters
-        ----------
-        hostname : string or list
-            The CAS host to connect to.
-        port : int
-            The CAS port to connect to.
-        protocol : string, optional
-            The protocol override value.
-
-        Returns
-        -------
-        string
-            'cas' or 'http'
-
-        '''
         if protocol is None:
             protocol = cf.get_option('cas.protocol')
 
-        if isinstance(hostname, six.string_types):
-            hostname = re.split(r'\s+', hostname.strip())
-
         # Try to detect the proper protocol
+
         if protocol == 'auto':
 
             import socket
+            import ssl
 
-#           for ptype in ['http', 'https']:
-            for host in hostname:
```

Comment:
This loop needs to remain because there could be multiple possible controllers.  The first active controller is the one that is tested.

Reply:
Done.